packages/lit-dev-content/samples/tutorials/utils.cjs.ts

```diff
@@ -0,0 +1,77 @@
+import fs from 'fs/promises';
```

Comment:
I think we should move all eleventy-related/node scripts related to tutorials into `packages/lit-dev-tools-cjs`. That's where we've putting eleventy plugins/script.

It seems like putting the eleventynode side of this inside of the `samples/tutorials` directory is complicating things a bit -- because so far, the `samples/` directory has only been for the actual samples content, rather than the eleventy infrastructure for it.

Reply:
Reasoning behind this was to centralize and decrease the number of locations you'd need to visit to author a new tutorial since Arthur was worried about complicating the authoring workflow. Currently to create one the locations you need to edit are:

1. `lit-dev-content/samples/tutorials/[dirname]`
2. `lit-dev-content/site/tutorials/tutorials.11tydata.js`
3. `lit-dev-content/site/tutorials/content/[dirname]`

My worries about moving it to `lit-dev-tools-cjs` are:

1. it adds another place the author must edit
2. These cjs files contain content themselves not tools as they are extensions of (`lit-dev-content/site/__data/*.js` which are also cjs files)

I agree with the TS causing more complexity. Instead I can make these `.js` files and add `//@ts-check` to still provide the TS typechecking for authorship. That should knock out the extra tsconfig.json and any changes I made to the build scripts

Reply:
Sorry, I didn't mean that data you would need to edit to make a tutorial should go there, just any infrastructure scripts/eleventy plugins etc. I too like the idea that all tutorial content should be in the same directory.

Reply:
Clarifying some of my earlier comments...

I'm thinking we should mirror how the Playground examples list generation works today, which is this:

1. Each `samples/examples` directory has a `project.json` which has the title/description/section metadata in it. Example: https://github.com/lit/lit.dev/blob/main/packages/lit-dev-content/samples/examples/directive-async-replace/project.json

2. There is a script in `lit-dev-content/site/_data/playground_examples.js` which scans for all of the `project.json` files and returns the data for Eleventy: https://github.com/lit/lit.dev/blob/main/packages/lit-dev-content/site/_data/playground_examples.js

3. The Playground page reads the `playground_examples` data for templating: https://github.com/lit/lit.dev/blob/787b789b41f1fbb360a05c296a232b1963f940c9/packages/lit-dev-content/site/playground/index.html#L22

The tutorials generation is really similar to this, so I think we should mirror the same pattern. Specifically:

1. Each `samples/tutorials` directory has a `tutorial.json` which contains the list of steps and other metadata related to the index view.

2. Add a new `lit-dev-content/site/_data/tutorials.js` script.

3. Consume the `tutorials` variable in the template for the new tutorials page.

Benefits of this:

- Simpler infrastructure. We won't need the second `tsc` pipeline in the samples directory.

- I think a declarative JSON approach for tutorial metadata is probably than each tutorial having its own Node script:

  - If we later support tutorials in gists, then it wouldn't be possible to read the metadata out of a `.js` files, because we'd need to `eval` it, which would be unsafe. 

  - It feels more conventional/natural. It's similar to the existing Playground `project.json` files we have.

  - I hear you on the type checking concern, but:
    - We'll be adding tutorials quite slowly so the risk seems pretty low to begin with.
    - We can include a validity check in the step 2 script. [typescript-json-schema](https://github.com/YousefED/typescript-json-schema) could be a good way to do this, because it will generate a JSON schema automatically from the TypeScript interface. We could then automatically check validity from the schema. JSON schema definitions are also understood by VSCode using the special `"$schema" property.